agent/verify_job.go

```diff
@@ -48,7 +49,10 @@ func (r *JobRunner) verifyJob(keySet jwk.Set) error {
 	}
 
 	// Verify the signature
-	if err := signature.Verify(step.Signature, r.conf.JWKS, r.conf.Job.Env, stepWithInvariants); err != nil {
+	if err := signature.Verify(ctx, step.Signature, r.conf.JWKS, stepWithInvariants,
+		signature.WithEnv(r.conf.Job.Env),
+		signature.WithLogger(r.agentLogger),
+		signature.WithDebugSigning(r.conf.AgentConfiguration.DebugSigning)); err != nil {
```

Comment:
[optional weakly-held opinion] This is big enough that I would lean towards separating the `if` statement and condition (and, since we're fine with function calls across multiple lines, spread out the args a bit more):

```suggestion
	err := signature.Verify(
		ctx, 
		step.Signature, 
		r.conf.JWKS, 
		stepWithInvariants,
		signature.WithEnv(r.conf.Job.Env),
		signature.WithLogger(r.agentLogger),
		signature.WithDebugSigning(r.conf.AgentConfiguration.DebugSigning),
	)
	if err != nil {
```